src/datachain/catalog/catalog.py

```diff
@@ -1931,19 +1931,37 @@ def query(
                 output=output,
             )
 
+        def _get_latest_dataset():
+            for dataset, dv, job in self.list_datasets_versions():
+                if job and str(job.id) == job_id:
+                    yield dv.created_at, (dataset, dv)
```

Comment:
We might need an API for listing dataset versions by a given job id in the future for performance reasons.